src/restruct_poc/math/quaternion_operations.hpp

```diff
@@ -18,18 +20,20 @@ KOKKOS_INLINE_FUNCTION void QuaternionToRotationMatrix(Q q, View_Rotation R) {
     R(2, 2) = q(0) * q(0) - q(1) * q(1) - q(2) * q(2) + q(3) * q(3);
 }
 
-template <typename Q, typename View1, typename View2>
-KOKKOS_INLINE_FUNCTION void QuaternionRotateVector(Q q, View1 v, View2 v_rot) {
-    v_rot[0] = (q(0) * q(0) + q(1) * q(1) - q(2) * q(2) - q(3) * q(3)) * v(0) +
+/// Rotates provided vector by provided *unit* quaternion and returns the result
+template <typename Quaternion, typename View1, typename View2>
+KOKKOS_INLINE_FUNCTION void RotateVectorByQuaternion(Quaternion q, View1 v, View2 v_rot) {
+    v_rot(0) = (q(0) * q(0) + q(1) * q(1) - q(2) * q(2) - q(3) * q(3)) * v(0) +
                2. * (q(1) * q(2) - q(0) * q(3)) * v(1) + 2. * (q(1) * q(3) + q(0) * q(2)) * v(2);
-    v_rot[1] = 2. * (q(1) * q(2) + q(0) * q(3)) * v(0) +
+    v_rot(1) = 2. * (q(1) * q(2) + q(0) * q(3)) * v(0) +
                (q(0) * q(0) - q(1) * q(1) + q(2) * q(2) - q(3) * q(3)) * v(1) +
                2. * (q(2) * q(3) - q(0) * q(1)) * v(2);
-    v_rot[2] = 2. * (q(1) * q(3) - q(0) * q(2)) * v(0) + 2. * (q(2) * q(3) + q(0) * q(1)) * v(1) +
+    v_rot(2) = 2. * (q(1) * q(3) - q(0) * q(2)) * v(0) + 2. * (q(2) * q(3) + q(0) * q(1)) * v(1) +
                (q(0) * q(0) - q(1) * q(1) - q(2) * q(2) + q(3) * q(3)) * v(2);
 }
 
-inline std::array<double, 3> QuaternionRotateVector(
+// TODO This is a duplicate of the above function. We should look into merging them.
```

Comment:
This (and the other array functions here) was added as a convenience function for our test code.  We use them for setting up our tests and computing initial values.  I think these functions are much better for that purpose (readability, ease of use with standard C++), while the others are better for use in our performance critical kernels.  Arrays and Views have different enough APIs and semantics that merging them doesn't make sense